src/main/java/dk/kb/labsapi/ImageExport.java

```diff
@@ -204,25 +214,31 @@ public List<IllustrationMetadata> getMetadataForIllustrations(QueryResponse solr
      * @param solrResponse to extract illustrations from.
      * @return a list of strings. Each string contains metadata for a single illustration from the input query response.
      */
-     private List<String> getIllustrationsList(QueryResponse solrResponse) {
+     public List<String> getIllustrationsList(QueryResponse solrResponse) {
         SolrDocumentList responseList = new SolrDocumentList();
         responseList = solrResponse.getResults();
         List<String> illustrationList = new ArrayList<>();
+
+        // Set for deduplication on pageUUID level
+        Set<String> uniqueUUIDs = new HashSet<>();
+
         // Extract metadata from documents in solr response
         for (int i = 0; i<responseList.size(); i++) {
             String pageUUID = responseList.get(i).getFieldValue("pageUUID").toString();
-            long pageWidth = (long) responseList.get(i).getFieldValue("page_width");
-            long pageHeight = (long) responseList.get(i).getFieldValue("page_height");
-            List<String> illustrations = (List<String>) responseList.get(i).getFieldValue("illustration");
-            // Check if illustrations are present. If not, continue to next SolrDocument in list
-            if (illustrations == null) {
-                continue;
-            }
-            // Create metadata string for each illustration
-            for (int j = 0; j< illustrations.size(); j++){
-                illustrations.set(j, illustrations.get(j) + "," + pageUUID + "," + pageWidth + "," + pageHeight);
+            if (uniqueUUIDs.add(pageUUID)){
+                long pageWidth = (long) responseList.get(i).getFieldValue("page_width");
+                long pageHeight = (long) responseList.get(i).getFieldValue("page_height");
+                List<String> illustrations = (List<String>) responseList.get(i).getFieldValue("illustration");
+                // Check if illustrations are present. If not, continue to next SolrDocument in list
+                if (illustrations == null) {
+                    continue;
+                }
+                // Create metadata string for each illustration
+                for (int j = 0; j< illustrations.size(); j++){
+                    illustrations.set(j, illustrations.get(j) + "," + pageUUID + "," + pageWidth + "," + pageHeight);
+                }
+                illustrationList.addAll(illustrations);
```

Comment:
Take a look at this code and let's talk about it. Not sure if it will help readability or muddy it:
```java
     public List<String> getIllustrationsListStreaming(QueryResponse solrResponse) {
         Set<Object> uniqueUUIDs = new HashSet<>();
         return solrResponse.getResults().stream().
                 filter(doc -> uniqueUUIDs.add(doc.getFieldValue("pageUUID"))).
                 flatMap(this::docToIllustrationStrings).
                 collect(Collectors.toList());
    }

    private Stream<String> docToIllustrationStrings(SolrDocument solrDoc) {
        String pageUUID = solrDoc.getFieldValue("pageUUID").toString();
        long pageWidth = (long) solrDoc.getFieldValue("page_width");
        long pageHeight = (long) solrDoc.getFieldValue("page_height");
        List<String> illustrations = (List<String>) solrDoc.getFieldValue("illustration");
        return illustrations == null ? Stream.empty() : 
                illustrations.stream().map(illStr -> illStr + "," + pageUUID + "," + pageWidth + "," + pageHeight);
    }
```

Reply:
For outsiders: We talked about this internally and decided to stick with the "classic" code style as we have not fully embraced the streaming style yet.